src/libsystemd-network/dhcp-option.c

```diff
@@ -35,6 +36,31 @@ static int option_append(uint8_t options[], size_t size, size_t *offset,
                 *offset += 1;
                 break;
 
+        case SD_DHCP_OPTION_USER_CLASS: {
+                size_t i = 0;
+                char **s;
+
+                STRV_FOREACH(s, (char **) optval)
+                        i += strlen(*s) + 1;
+
+                if (size < *offset + i + 2)
+                        return -ENOBUFS;
+
+                options[*offset] = code;
+                options[*offset + 1] =  i;
+                *offset += 2;
+
+                STRV_FOREACH(s, (char **) optval) {
+                        i = strlen(*s);
+
+                        options[*offset] = i;
+
+                        memcpy_safe(&options[*offset + 1], *s, i);
+                        *offset += i + 1;
+                }
+
+                break;
+        }
```

Comment:
actually, i think it's strange to stuff this into option_append(), and then sometimes consider "optval" a simple byte array and other times something more generic. That's not typesafe, and C is really about typesafety.

Please add a wrapper maybe, called option_append_userclass() which internally calls option_append() but first formats the strv into a linearized blob...

Reply:
I think this will duplicate work but let me try

Reply:
I figure the user class is something different. First in the header we put the total length and the code 77. then each string with length is added. option_append adds the string with code.